pyfar/classes/transmission_matrix.py

```diff
@@ -0,0 +1,610 @@
+    the form of 1/Zl (as admittance) in this case to avoid such problems.
+    Nevertheless, there are additional cases where certain entries of the
+    T-matrix being zero might still lead to the main denominator becoming zero.
+    For these cases, the denominator is set to eps.
+
+    References
+    ----------
+    .. [#] M. Lampton. “Transmission Matrices in Electroacoustics". Acustica.
+           Vol. 39, pp. 239-251. 1978.
+
+    """
+
+    def __init__(self, data, frequencies, comment = ""):
+        """Create TransmissionMatrix with data, and frequencies.
+
+        To ensure handing the correct data, it is recommended to use the
+        :py:func:`~from_abcd` method to create objects.
+
+        Parameters
+        ----------
+        data : array, double
+            Raw data in the frequency domain. The memory layout of Data is 'C'.
+            In contrast to the :py:class:`~pyfar.classes.audio.FrequencyData`
+            class, data must have a shape of the form (..., 2, 2, N), e.g.
+            (2, 2, 1024) or (3, 2, 2, 1024). In those examples 1024 refers to
+            the number of frequency bins and the two dimensions before that to
+            the (2x2) ABCD matrices. Supported data types are ``int``,
+            ``float`` or ``complex``, where ``int`` is converted to ``float``.
+        frequencies : array, double
+            Frequencies of the data in Hz. The number of frequencies must match
+            the size of the last dimension of data.
+        comment : str, optional
+            A comment related to the data. The default is ``""``, which
+            initializes an empty string.
+
+        """
+        shape = np.shape(data)
+        n_dim = len(shape)
+        if n_dim < 3 or shape[-3] != 2 or shape[-2] != 2:
+            raise ValueError("'data' must have a shape like [..., 2, 2, N]"
+                             ", e.g. [2, 2, 100].")
+
+        super().__init__(data, frequencies, comment)
+
+    @classmethod
+    def from_abcd(cls, A,B,C,D, frequencies = None):
+        """Create a TransmissionMatrix object from A-, B-, C-, D-data, and
+        frequencies.
+
+        Parameters
+        ----------
+        A, B, C, D : FrequencyData, array, double
+            Raw data for the matrix entries A, B, C and D. Must use the same
+            data type for all inputs.
+        frequencies : array, double, None
+            Frequencies of the data in Hz. This is optional if using the
+            FrequencyData type for A, B, C, D.
+
+        """
+        if (
+            not isinstance(B, type(A))
+            or not isinstance(C, type(A))
+            or not isinstance(D, type(A))
+        ):
+            raise ValueError("A-,B-,C- and D-Matrices must "
+                             "be of the same type.")
+
+        if isinstance(A, FrequencyData):
+            frequencies = A.frequencies
+            A = A.freq
+            B = B.freq
+            C = C.freq
+            D = D.freq
```

Comment:
```suggestion
            frequencies = A.frequencies
            for d in (B,C,D):
                if isinstance(d, FrequencyData):
                    assert npt.allclose(
                        A.frequencies, d.frequencies, atol=1e-15)
                else:
                    raise ValueError(
                        "If using FrequencyData objects for A, "
                        "B, C, D, all must be FrequencyData objects.")
    
```
I think we need to make sure that all object have the same Frequency bins.